src/cpu/aarch64/brgemm/jit_brgemm_kernel.cpp

```diff
@@ -518,14 +518,7 @@ void jit_brgemm_kernel_t::cvt2ps(data_type_t type_in, const ZReg zmm_in,
             break;
         }
         case data_type::s8:
-            if (mask_flag)
-                set_preg(P_TMP.b, brg.ldb_tail, X_TMP_0, X_TMP_1);
-            else
-                set_preg(P_TMP.b, brg.ld_block, X_TMP_0, X_TMP_1);
-            add_imm(X_DEFAULT_ADDR, addr, offset - base_offset, X_TMP_0);
-            ld1b(z_tmp_1().b, P_TMP / T_z, ptr(X_DEFAULT_ADDR));
-            sunpklo(z_tmp_1().h, z_tmp_1().b);
-            sunpklo(z_tmp_1().s, z_tmp_1().h);
+            LD_MUL_VL(ld1sb, z_tmp_1().s, mask, addr, offset - base_offset, 1);
```

Comment:
Minor: given the amount of recent contributions to brgemm, I'd highly recommend to turn this macro into a function of `jit_generator` or a standalone function that takes `jit_generator` by pointer to form a collection.

Macro is undebuggable and not reusable given there are already 4 version of it across aarch64 codebase.

Reply:
> not only leads to compilation errors due to mismatched definitions...

I might be wrong by C++ templating usually handles that pretty well if one instruction corresponds to a single usage criterion. I'm not sure I see the point of being it more complex without a code example of now versus would be.